src/api/app/controllers/trigger_controller.rb

```diff
@@ -77,9 +85,19 @@ def require_valid_token
 
     raise NoPermissionForInactive unless User.session.is_active?
 
-    @pkg = @token.package || Package.get_by_project_and_name(params[:project].to_s, params[:package].to_s, use_source: true)
-
-    raise ActiveRecord::RecordNotFound unless @pkg
-    raise NoPermissionForPackage.setup('no_permission', 403, "no permission for package #{@pkg} in project #{@pkg.project}") unless policy(@pkg).update?
+    if @token.package
+      @pkg = @token.package
+      @prj = @pkg.project
+    else
+      @prj = Project.get_by_name(params[:project])
+      if @token.class == Token::Rebuild
```

Comment:
so the rebuild token is for a project and not for a package?

Reply:
it can be for a source package object

but it can be also global and then you need to check it for project where the package build container lives.

As discussed face to face: It would be better to have a package build container object and use policies for that. This would also help in other controllers and operations. But it is outside of the scope here...